packages/foundry/contracts/prediction-market/lib/PredictionMarketLib.sol

```diff
@@ -0,0 +1,297 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { 
+    PredictionMarket,
+    Side 
+} from '../Types.sol';
+
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+library PredictionMarketLib { 
+
+    /**
+     * @notice Prediction market has zero token balance for one of both pair tokens
+     */
+    error ZeroTokenBalance();
+
+    /**
+     * @notice Prediction market has insufficent liquidity to execute swap
+     */
+    error CannotSwapInMarket();
+
+    /**
+     * @notice Prediction market does not have liquidity on both sides
+     * @dev Occurs when a user tries to add liquidity proportionally to a single sided market
+     * In this case, the market cannot quote the side with zero liquidity
+     */
+     error CannotAddProportionalLiquidityToSingleSidedMarket();
+
+    
+    /**
+     * @notice Quote the current prices (probabilities) of the prediction market
+     * @dev The total value of the market is always equal to the deposited liquidity, thus to
+     * get the probability or price of each outcome, we need to divide the net outcome balance
+     * by the total deposited liquidity.
+     * @param self The prediction market to quote
+     */
+    function quote(
+        PredictionMarket memory self
+    ) internal pure returns (uint256 quoteBull, uint256 quoteBear) {
+        uint256 liquidity = netLiquidity(self);
+
+        quoteBull = Math.mulDiv(self.balanceBull, 1e18, liquidity);
```

Comment:
Could use our FixedPoint library. If self.balanceBull is FP 18, can just say `self.balanceBull.divDown(liquidity);` (including and using FixedPoint). You're importing it, but don't seem to be using it. (It would mean thinking about rounding directions.)

We have a naming convention for FP18 (Scaled18) and native token decimals (Raw), and use it consistently for state variables and parameters, which is self-documenting and makes it easier to understand.

Reply:
Thanks for that, I wasn't aware of that opportunity